File: JS/bubblesort.js
//helper function to swap elements
function swap(x, y, arr){
    let temp=arr[x];
    arr[x]=arr[y];
    arr[y]=temp;
}

function bubbleSort(array){
    //iterate over the length of the array
    for(let i=0; i<array.length; i++){
        //iterate over all unbubbled elements
        for(let j=0; j<array.length-1-i; j++){
            //if the lower value is greater, swap
            if(array[j]>array[j+1]){
                swap(j,j+1,array);
            }
        }
    }
    return array;
}

File: Python/bubble_sort.py
# this program implements a bubblesort algorithm
# in bubblesort, each pair of adjacent elements is compared and the elements are swapped if they are not in the correct order. this continues until everything is properly ordered

def bubble_sort(list):
    for pair in range(len(list)-1, 0, -1): #iterate over each pair in the list
        for i in range(pair): #iterate over each item in the pair
            if(list[i]>list[i+1]): #if unordered, create temp, switch elements
                temp = list[i]
                list[i] = list[i+1]
                list[i+1] = temp

#example list to test
list = [19,2,31,45,6,11,121,27]
bubble_sort(list)
print(list)